app/src/main/java/com/doctor/yumyum/presentation/ui/write/viewmodel/WriteTagViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.doctor.yumyum.presentation.ui.write.viewmodel
+
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.doctor.yumyum.common.base.BaseViewModel
+
+class WriteTagViewModel : BaseViewModel(){
+    private val _guideText : MutableLiveData<String> = MutableLiveData()
+    val guideText : LiveData<String>
+        get() = _guideText
+
+    val etTagItem : MutableLiveData<String> = MutableLiveData()
+
+    private val _rvTagItem : MutableLiveData<String> = MutableLiveData()
+    val rvTagItem : LiveData<String>
+        get() = _rvTagItem
+
+
+    fun setGuideText(type : String){
+        _guideText.value = type
+    }
+
+    fun setTagItem(){
+        val recentTagItem = this.etTagItem.value.also {
+            if (it.isNullOrBlank()) {
+                toast("재료를 입력해주세요")
+                return
+            }
+        }
```

Comment:
```suggestion
            if (etTagItem.value.isNullOrBlank() == true) {
                toast("재료를 입력해주세요")
                return
            }
            _rvTagItem.value = etTagItem.value
        }
```
1. 사실 제가 제안한 방법보다, _rvTagItem / etTagItem 둘 중 하나만 사용해도 괜찮을 것 같습니다 : ) 두 변수가 xml에서 사용하는 목적이 동일한 것 같아요.
2. scope function 은 잘못 사용한다면 가독성을 해칠 수 있는 아이라고 생각해요. 약간은 풀어쓰는 것도 좋을 것 같아 제안드립니다 ㅎㅎ
2. setter 여서 파라미터를 통해 변수가 셋팅되는 것 같았는데, 아닌 것 같아 읽는데 약간 당황스러웠던 것 같아요.`setTagItem(tag: String) {  }` 혹은 validate, check... 정도는 어떨까요? (그럼에도 `_rvTagItem.value = recentTagItem` 요녀석이 맘이 걸리긴 함...👀)

Reply:
1. 좀 더 고민해보니까 rvTagItem을 String에서 List로 변경해서 사용해야 다른 기능에서도 사용이 가능 할 것 같아요. 두 변수의 목적을 다르게 변경하겠습니다😀
2. 네!!!
3. 위에 비어있는 값인지 확인하는 부분에서 유효성 검사를 하고 `recentTagItem`를 `setTagItem`의 파라미터로 전달하는 방향으로 바꾸면 될까요?!